cli/npm/managed/resolvers/local.rs

```diff
@@ -508,6 +660,73 @@ async fn sync_resolution_with_fs(
     }
   }
 
+  if !packages_with_scripts.is_empty() {
+    // get custom commands for each bin available in the node_modules dir (essentially
+    // the scripts that are in `node_modules/.bin`)
+    let base = resolve_baseline_custom_commands(
+      snapshot,
+      &package_partitions.packages,
+      &deno_local_registry_dir,
+    )?;
+
+    for (package, package_path, scripts_run_path) in packages_with_scripts {
+      // add custom commands for binaries from the package's dependencies. this will take precedence over the
+      // baseline commands, so if the package relies on a bin that conflicts with one higher in the dependency tree, the
+      // correct bin will be used.
+      let custom_commands = resolve_custom_commands_from_deps(
+        base.clone(),
+        &package,
+        snapshot,
+        &deno_local_registry_dir,
+      )?;
+      for script_name in ["preinstall", "install", "postinstall"] {
+        if let Some(script) = package.scripts.get(script_name) {
+          let exit_code =
+            crate::task_runner::run_task(crate::task_runner::RunTaskOptions {
+              task_name: script_name,
+              script,
+              cwd: &package_path,
+              env_vars: crate::task_runner::real_env_vars(),
+              custom_commands: custom_commands.clone(),
+              // todo: wrong
```

Comment:
Can you expand on this comment about what this value needs to be? Is it `options.initial_cwd()`?

Reply:
Oh yeah I forgot about that todo, yup it should be `options.initial_cwd()`. I just need to thread it through somewhere (or I guess I could just do `std::env::current_dir()`)

Reply:
Ok I just bundled the allowed scripts + initial cwd into a little struct, so it just gets threaded through with the allowed scripts. So this should be correct now